### PR TITLE
fix status-bar font color

### LINKED ISSUE #15590: Status bar font turns white after background or opening a chat
# Bug Report

## Problem 

If the device is in light mode, then the status bar becomes invisible on the home screens after the background or opening a chat.
Relogin fixes the issue.

**Steps:**
1. Open Status
2. Send the app to the background 

https://user-images.githubusercontent.com/67952253/230332752-1275d923-d618-47fd-879c-f6ecdf1ae9d9.mp4

**or**
3. Open a chat/community

https://user-images.githubusercontent.com/67952253/230333430-c1baca3b-947d-447e-aae1-a9e33d32c240.mp4

### Additional Information

- Status version: RC 1.22
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes: https://github.com/status-im/status-mobile/issues/15590

### Summary
- We are initializing root before processing settings data (which has the account theme)
- RNN Bug - We are also changing the status-bar-theme at runtime, but for some reason theme used at the time of root initialization has high priority and gets restored on navigation (going back from chat, etc.)
- Also we need to change app theme, before root

### Testing
- The issue should be fixed for both light and dark app themes, but as mentioned above, the theme needs to be provided at the time of root initialization. So if the theme changes after login, then the behavior will be the same, until the account relogin.
- PR also makes alters small modification made in https://github.com/status-im/status-mobile/pull/15574. So please check pairing is working as expected

status: ready

### CODE REVIEW

**Comment by churik** [comment]
https://user-images.githubusercontent.com/4557972/230419939-50f4b856-fdc1-42ee-aa24-d3fca860f4f5.mp4

@Parveshdhull there is another issue with "Jump to" screen in light mode - the bar is not visible (both IOS and Android)
We can test it separately though, it is not as visible as the initial issue.


Testing on `release 1.22` branch

**Comment by Parveshdhull** [comment]
hi @churik Thank you for testing the PR, for ios behavior is expected due to bug in rnn (https://github.com/status-im/status-mobile/issues/15260). For android it should be fixed in https://github.com/status-im/status-mobile/pull/15599

PS: Issue is only in release branch

**Review by jo-mut — APPROVED** [review]

**Comment by qoqobolo** [comment]
Hi @Parveshdhull, thanks for the PR!

Can ask you a question about a similar issue with the status bar on iOS? It is also reproducible in develop, and I see that there are some more status bar issues to be fixed, so I'm not sure this one in the scope of this PR.
But could you tell for sure?

**ISSUE 1: Status bar has a white font on the home screen on iOS with light theme for a fresh install**

Screenshot 2023-04-14 at 11 05 18


**Steps:**
1. Set a light theme on your iOS device 
2. Install Status
3. Create an account
4. Check the home screen

Relogin/reopening the app doesn't help in this case.
Should we log this issue separately?

**Comment by Parveshdhull** [comment]
Hi @qoqobolo, Thank you for testing the PR.
I think it will be best if we can fix this issue in the scope of PR.
I will see what is causing it and will let you know once it is fixed.

**Comment by Parveshdhull** [comment]
Hi @qoqobolo, Thank you for finding the issue. It should be fixed now

**Comment by qoqobolo** [comment]
Thanks for the fix @Parveshdhull 
Looks good to me, let's wait for @flexsurfer's review. 
PR can be merged once approval is received unless re-testing is needed.

**Review by flexsurfer — APPROVED** [review]

**Pull request merged**
